Index: gnu/llvm/lib/CodeGen/StackProtector.cpp
===================================================================
RCS file: /cvs/src/gnu/llvm/lib/CodeGen/StackProtector.cpp,v
retrieving revision 1.8
diff -u -p -r1.8 StackProtector.cpp
--- gnu/llvm/lib/CodeGen/StackProtector.cpp	23 Jun 2019 22:05:12 -0000	1.8
+++ gnu/llvm/lib/CodeGen/StackProtector.cpp	7 Jul 2019 16:55:30 -0000
@@ -322,8 +322,10 @@ bool StackProtector::RequiresStackProtec
 static Value *getStackGuard(const TargetLoweringBase *TLI, Module *M,
                             IRBuilder<> &B,
                             bool *SupportsSelectionDAGSP = nullptr) {
-  if (Value *Guard = TLI->getIRStackGuard(B))
+  if (Value *Guard = TLI->getIRStackGuard(B)) {
+    TLI->insertSSPDeclarations(*M);
     return B.CreateLoad(Guard, true, "StackGuard");
+  }
 
   // Use SelectionDAG SSP handling, since there isn't an IR guard.
   //
