Index: lib/csu/arm/md_init.h
===================================================================
RCS file: /cvs/src/lib/csu/arm/md_init.h,v
retrieving revision 1.18
diff -u -p -r1.18 md_init.h
--- lib/csu/arm/md_init.h	18 Nov 2023 16:26:16 -0000	1.18
+++ lib/csu/arm/md_init.h	2 Feb 2024 22:33:59 -0000
@@ -55,6 +55,28 @@
 	"	.int "#value"						\n" \
 	"	.previous")
 
+#if defined(CPU_ARMv6)
+
+#ifdef __PIC__
+	/* This nastyness derived from gcc3 output */
+#define MD_SECT_CALL_FUNC(section, func) \
+	__asm (".section "#section", \"ax\"		\n" \
+	"	bl " #func "(PLT)			\n" \
+	"	.previous")
+#else /* !__PIC__ */
+#define MD_SECT_CALL_FUNC(section, func) \
+	__asm (".section "#section", \"ax\"	\n" \
+	"	adr r0, 1f			\n" \
+	"	ldr r0, [r0]			\n" \
+	"	adr lr, 2f			\n" \
+	"	mov pc,	r0			\n" \
+	"1:	.word " #func "			\n" \
+	"2:					\n" \
+	"	.previous")
+#endif /* !__PIC__ */
+
+#else /* !CPU_ARMv6 */
+
 #define MD_SECT_CALL_FUNC(section, func) \
 	__asm (".section "#section", \"ax\"		\n" \
 	"	movw	r0, #:lower16:" #func "- 1f - 8	\n" \
@@ -62,6 +84,8 @@
 	"1:	add	r0, r0, pc			\n" \
 	"	blx	r0				\n" \
 	"	.previous")
+
+#endif  /* CPU_ARMv6 */
 
 #define MD_SECTION_PROLOGUE(sect, entry_pt)	\
 	__asm (					\
