Index: Makefile.cross
===================================================================
RCS file: /cvs/src/Makefile.cross,v
retrieving revision 1.111
diff -u -p -r1.111 Makefile.cross
--- Makefile.cross	3 May 2021 00:04:31 -0000	1.111
+++ Makefile.cross	2 Feb 2024 22:33:50 -0000
@@ -43,8 +43,9 @@ MACHINE_IS_LP64 =
 
 #CROSSCPPFLAGS?=	-nostdinc -I${CROSSDIR}/usr/include
 #CROSSLDFLAGS?=	-nostdlib -L${CROSSDIR}/usr/lib -static
-CROSSCFLAGS?=	${CROSSCPPFLAGS} -O2 ${PIPE} ${DEBUG}
-CROSSCXXFLAGS?=	${CROSSCPPFLAGS}
+#CROSSCFLAGS?=	${CROSSCPPFLAGS} -O2 ${PIPE} ${DEBUG} -marm -mfpu=vfp -mcpu=arm1176jzf-s -mtune=arm1176jzf-s -mfloat-abi=softfp
+CROSSCFLAGS?=	${CROSSCPPFLAGS} -O0 -g ${PIPE} ${DEBUG} -march=armv6 -DCPU_ARMv6
+CROSSCXXFLAGS?=	${CROSSCPPFLAGS} -march=armv6 -DCPU_ARMv6
 #LDSTATIC?=	-static
 
 CROSSDIR=	${DESTDIR}/usr/cross/${TARGET}
