Index: share/man/man4/Makefile
===================================================================
RCS file: /cvs/src/share/man/man4/Makefile,v
retrieving revision 1.736
diff -u -p -r1.736 Makefile
--- share/man/man4/Makefile	7 Oct 2019 20:10:21 -0000	1.736
+++ share/man/man4/Makefile	8 Oct 2019 21:21:27 -0000
@@ -70,8 +70,8 @@ MAN=	aac.4 abcrtc.4 ac97.4 acphy.4 acrtc
 	sf.4 sili.4 simplefb.4 siop.4 sis.4 sk.4 sli.4 \
 	sm.4 smsc.4 softraid.4 spdmem.4 sdtemp.4 speaker.4 sppp.4 sqphy.4 \
 	ssdfb.4 st.4 ste.4 stge.4 sti.4 stp.4 sv.4 switch.4 sxiccmu.4 \
-	sximmc.4 sxipio.4 sxirsb.4 sxirtc.4 sxisyscon.4 sxitemp.4 sxitwi.4 \
-	sym.4 sypwr.4 syscon.4 \
+	sximmc.4 sxipio.4 sxirsb.4 sxirtc.4 sxisid.4 sxisyscon.4 sxitemp.4 \
+	sxitwi.4 sym.4 sypwr.4 syscon.4 \
 	tcic.4 tcp.4 termios.4 tht.4 ti.4 tipmic.4 tl.4 \
 	tlphy.4 thmc.4 tpm.4 tpmr.4 tqphy.4 trm.4 trunk.4 tsl.4 tty.4 \
 	tun.4 tap.4 twe.4 \
