Index: sys/dev/pwm/pwmvar.h
===================================================================
RCS file: sys/dev/pwm/pwmvar.h
diff -N sys/dev/pwm/pwmvar.h
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ sys/dev/pwm/pwmvar.h	28 Aug 2019 22:16:40 -0000
@@ -0,0 +1,32 @@
+/*	$OpenBSD$	*/
+
+struct pwm_config {
+	uint16_t pc_period;
+	uint16_t pc_duty_cycle;
+};
+
+typedef struct pwm_controller {
+	void		*pc_cookie;
+	int		(*pc_enable)(void *);
+	int		(*pc_disable)(void *);
+	int		(*pc_get_config)(void *, struct pwm_config *);
+	int		(*pc_set_config)(void *, struct pwm_config *);
+
+} *pwm_tag_t;
+
+struct pwm_attach_args {
+	pwm_tag_t	pa_tag;
+	char		*pa_name;
+	void		*pa_cookie;
+
+	uint8_t	pa_channel;
+};
+
+#define pwm_enable(sc)						\
+	(*(sc)->pc_enable)((sc)->pc_cookie)
+#define pwm_disable(sc)					\
+	(*(sc)->pc_disable)((sc)->pc_cookie)
+#define pwm_get_config(sc, config)				\
+	(*(sc)->pc_get_config)((sc)->pc_cookie, (config))
+#define pwm_set_config(sc, config)				\
+	(*(sc)->pc_set_config)((sc)->pc_cookie, (config))
