Index: sys/arch/arm/cortex/cortex.c
===================================================================
RCS file: /cvs/src/sys/arch/arm/cortex/cortex.c,v
retrieving revision 1.7
diff -u -p -r1.7 cortex.c
--- sys/arch/arm/cortex/cortex.c	12 Mar 2022 14:40:41 -0000	1.7
+++ sys/arch/arm/cortex/cortex.c	2 Feb 2024 22:34:19 -0000
@@ -103,6 +103,12 @@ cortexmatch(struct device *parent, void 
 	if (strcmp(cf->cf_driver->cd_name, ma->ma_name) != 0)
 		return (0);
 
+#if defined(CPU_ARMv6)
+	if ((cputype & CPU_ID_CORTEX_A6_MASK) == CPU_ID_CORTEX_A6) {
+	        if (armv6_periphbase())
+			return (1);	  
+	}
+#elif defined(CPU_ARMv7)
 	if ((cputype & CPU_ID_CORTEX_A7_MASK) == CPU_ID_CORTEX_A7 ||
 	    (cputype & CPU_ID_CORTEX_A9_MASK) == CPU_ID_CORTEX_A9 ||
 	    (cputype & CPU_ID_CORTEX_A15_MASK) == CPU_ID_CORTEX_A15 ||
@@ -110,6 +116,7 @@ cortexmatch(struct device *parent, void 
 		if (armv7_periphbase())
 			return (1);
 	}
+#endif /* CPU_ARMv7 */
 
 	return (0);
 }
@@ -137,8 +144,11 @@ cortexsearch(struct device *parent, void
 	ca.ca_name = cf->cf_driver->cd_name;
 	ca.ca_iot = &armv7_bs_tag;
 	ca.ca_dmat = &cortex_bus_dma_tag;
+#if defined(CPU_ARMv7)
 	ca.ca_periphbase = armv7_periphbase();
-
+#elif defined(CPU_ARMv6)
+	ca.ca_periphbase = armv6_periphbase();
+#endif
 	/* allow for devices to be disabled in UKC */
 	if ((*cf->cf_attach->ca_match)(parent, cf, &ca) == 0)
 		return 0;
