Index: sys/dev/fdt/sxiccmu_clocks.h
===================================================================
RCS file: /cvs/src/sys/dev/fdt/sxiccmu_clocks.h,v
retrieving revision 1.24
diff -u -p -r1.24 sxiccmu_clocks.h
--- sys/dev/fdt/sxiccmu_clocks.h	12 Feb 2019 21:34:11 -0000	1.24
+++ sys/dev/fdt/sxiccmu_clocks.h	1 Aug 2019 07:57:17 -0000
@@ -140,9 +140,12 @@ struct sxiccmu_ccu_bit sun8i_a23_gates[]
 
 /* A64 */
 
+#define A64_CLK_PLL_CPUX	1
+
 #define A64_CLK_PLL_PERIPH0	11
 #define A64_CLK_PLL_PERIPH0_2X	12
 
+#define A64_CLK_CPUX		21
 #define A64_CLK_AXI		22
 #define A64_CLK_APB		23
 #define A64_CLK_AHB1		24
