Index: sys/dev/fdt/files.fdt
===================================================================
RCS file: /cvs/src/sys/dev/fdt/files.fdt,v
retrieving revision 1.101
diff -u -p -r1.101 files.fdt
--- sys/dev/fdt/files.fdt	21 Oct 2019 20:52:33 -0000	1.101
+++ sys/dev/fdt/files.fdt	23 Oct 2019 01:44:29 -0000
@@ -44,6 +44,10 @@ device	sxisyscon: fdt
 attach	sxisyscon at fdt
 file	dev/fdt/sxisyscon.c		sxisyscon
 
+device	sxisid: fdt
+attach	sxisid at fdt
+file	dev/fdt/sxisid.c		sxisid
+
 device	sxitemp
 attach	sxitemp at fdt
 file	dev/fdt/sxitemp.c		sxitemp
@@ -55,6 +59,10 @@ file	dev/fdt/sxits.c			sxits
 device	sxitwi: i2cbus
 attach	sxitwi at fdt
 file	dev/fdt/sxitwi.c		sxitwi
+
+device	sxiuphy
+attach	sxiuphy at fdt
+file	dev/fdt/sxiuphy.c		sxiuphy
 
 device	axppmic
 attach	axppmic at i2c
