Index: sys/arch/arm64/conf/RAMDISK
===================================================================
RCS file: /cvs/src/sys/arch/arm64/conf/RAMDISK,v
retrieving revision 1.105
diff -u -p -r1.105 RAMDISK
--- sys/arch/arm64/conf/RAMDISK	17 Oct 2019 22:32:06 -0000	1.105
+++ sys/arch/arm64/conf/RAMDISK	23 Oct 2019 01:44:26 -0000
@@ -107,6 +107,7 @@ vioscsi*	at virtio?
 
 simplefb*	at fdt?
 wsdisplay*	at simplefb?
+pwmbl*		at fdt?	# PWM backlight
 
 # iMX
 imxccm*		at fdt? early 1
@@ -189,6 +190,8 @@ sximmc*		at fdt?		# SD/MMC card controll
 sdmmc*		at sximmc?	# SD/MMC bus
 sxisyscon*	at fdt? early 1	# System controller
 sxitwi*		at fdt?		# I2C controller
+sxipwm*		at fdt?		# PWM controller
+sxiuphy*	at fdt?		# USB Phy
 iic*		at sxitwi?	# I2C bus
 dwxe*		at fdt?
 
