Index: sys/arch/arm64/conf/RAMDISK
===================================================================
RCS file: /cvs/src/sys/arch/arm64/conf/RAMDISK,v
retrieving revision 1.103
diff -u -p -r1.103 RAMDISK
--- sys/arch/arm64/conf/RAMDISK	9 Sep 2019 20:00:27 -0000	1.103
+++ sys/arch/arm64/conf/RAMDISK	12 Sep 2019 23:48:45 -0000
@@ -107,6 +107,7 @@ vioscsi*	at virtio?
 
 simplefb*	at fdt?
 wsdisplay*	at simplefb?
+pwmbl*			at pwm?		# PWM backlight
 
 # iMX
 imxccm*		at fdt? early 1
@@ -185,7 +186,9 @@ sxirtc*		at fdt? early 1	# Real Time Clo
 sximmc*		at fdt?		# SD/MMC card controller
 sdmmc*		at sximmc?	# SD/MMC bus
 sxisyscon*	at fdt? early 1	# System controller
+sxisid*		at fdt? early 1	# SID ROM
 sxitwi*		at fdt?		# I2C controller
+sxipwm*		at fdt? early 1		# PWM controller
 iic*		at sxitwi?	# I2C bus
 dwxe*		at fdt?
 
