Index: sys/arch/arm/arm/cpuswitch7.S
===================================================================
RCS file: /cvs/src/sys/arch/arm/arm/cpuswitch7.S,v
retrieving revision 1.18
diff -u -p -r1.18 cpuswitch7.S
--- sys/arch/arm/arm/cpuswitch7.S	24 Oct 2023 13:20:09 -0000	1.18
+++ sys/arch/arm/arm/cpuswitch7.S	2 Feb 2024 22:34:07 -0000
@@ -234,5 +234,5 @@ ENTRY(proc_trampoline)
 	PULLFRAME
 
 	movs	pc, lr			/* Exit */
-	dsb	nsh
-	isb
+//	dsb	nsh
+//	isb
