Index: sys/dev/fdt/files.fdt
===================================================================
RCS file: /cvs/src/sys/dev/fdt/files.fdt,v
retrieving revision 1.95
diff -u -p -r1.95 files.fdt
--- sys/dev/fdt/files.fdt	30 Sep 2019 20:44:13 -0000	1.95
+++ sys/dev/fdt/files.fdt	8 Oct 2019 20:50:06 -0000
@@ -36,6 +36,10 @@ device	sxisyscon: fdt
 attach	sxisyscon at fdt
 file	dev/fdt/sxisyscon.c		sxisyscon
 
+device	sxisid: fdt
+attach	sxisid at fdt
+file	dev/fdt/sxisid.c		sxisid
+
 device	sxitemp
 attach	sxitemp at fdt
 file	dev/fdt/sxitemp.c		sxitemp
@@ -47,6 +51,18 @@ file	dev/fdt/sxits.c			sxits
 device	sxitwi: i2cbus
 attach	sxitwi at fdt
 file	dev/fdt/sxitwi.c		sxitwi
+
+device	sxiuphy
+attach	sxiuphy at fdt
+file	dev/fdt/sxiuphy.c		sxiuphy
+
+device	sxipwm
+attach	sxipwm at fdt
+file	dev/fdt/sxipwm.c		sxipwm
+
+device	pwmbl
+attach	pwmbl at fdt
+file	dev/fdt/pwmbl.c		pwmbl
 
 device	axppmic
 attach	axppmic at i2c
