Index: sys/arch/arm64/conf/GENERIC
===================================================================
RCS file: /cvs/src/sys/arch/arm64/conf/GENERIC,v
retrieving revision 1.98
diff -u -p -r1.98 GENERIC
--- sys/arch/arm64/conf/GENERIC	1 Apr 2019 08:46:16 -0000	1.98
+++ sys/arch/arm64/conf/GENERIC	13 Apr 2019 00:52:48 -0000
@@ -172,6 +172,7 @@ axppmic*	at rsb?
 sxirtc*		at fdt?		# Real Time Clock
 sximmc*		at fdt?		# SD/MMC card controller
 sdmmc*		at sximmc?	# SD/MMC bus
+sxisid*		at fdt? early 1	# SID ROM  
 sxitemp*	at fdt?		# Temperature sensor
 sxitwi*		at fdt?		# I2C controller
 iic*		at sxitwi?	# I2C bus
