Index: sys/arch/armv7/conf/GENERIC
===================================================================
RCS file: /cvs/src/sys/arch/armv7/conf/GENERIC,v
retrieving revision 1.142
diff -u -p -r1.142 GENERIC
--- sys/arch/armv7/conf/GENERIC	12 Mar 2023 10:50:06 -0000	1.142
+++ sys/arch/armv7/conf/GENERIC	14 Feb 2024 11:30:02 -0000
@@ -220,6 +220,11 @@ cduart*		at fdt?
 zqclock*	at fdt?
 zqreset*	at fdt?
 
+# STM32MP1
+stmuart*	at fdt?
+stmdog*		at fdt?			# watchdog timer
+stmpwr*		at fdt?
+
 # I2C devices
 abcrtc*		at iic?			# Abracon x80x RTC
 axppmic*	at iic?			# axp209 pmic
