Index: sys/arch/arm/arm/irq_dispatch.S
===================================================================
RCS file: /cvs/src/sys/arch/arm/arm/irq_dispatch.S,v
retrieving revision 1.17
diff -u -p -r1.17 irq_dispatch.S
--- sys/arch/arm/arm/irq_dispatch.S	8 Dec 2022 01:25:44 -0000	1.17
+++ sys/arch/arm/arm/irq_dispatch.S	2 Feb 2024 22:34:10 -0000
@@ -128,8 +128,8 @@ ASENTRY_NP(irq_entry)
 	DO_AST
 	PULLFRAMEFROMSVCANDEXIT
 	movs	pc, lr			/* Exit */
-	dsb	nsh
-	isb
+//	dsb	nsh
+//	isb
 
 	.bss
 	.align	2
