Index: sys/arch/arm64/conf/RAMDISK
===================================================================
RCS file: /cvs/src/sys/arch/arm64/conf/RAMDISK,v
retrieving revision 1.88
diff -u -p -r1.88 RAMDISK
--- sys/arch/arm64/conf/RAMDISK	7 Jun 2019 04:03:04 -0000	1.88
+++ sys/arch/arm64/conf/RAMDISK	2 Aug 2019 17:37:22 -0000
@@ -170,8 +170,10 @@ sxirsb*		at fdt? early 1	# Reduced Seria
 sxirtc*		at fdt?		# Real Time Clock
 sximmc*		at fdt?		# SD/MMC card controller
 sdmmc*		at sximmc?	# SD/MMC bus
+sxisid*		at fdt? early 1	# SID ROM
 sxitwi*		at fdt?		# I2C controller
 iic*		at sxitwi?	# I2C bus
+sxiuphy*	at fdt?		# USB Phy
 dwxe*		at fdt?
 
 # PCI
