Index: sys/dev/fdt/simplefb.c
===================================================================
RCS file: /cvs/src/sys/dev/fdt/simplefb.c,v
retrieving revision 1.5
diff -u -p -r1.5 simplefb.c
--- sys/dev/fdt/simplefb.c	27 Aug 2018 09:30:07 -0000	1.5
+++ sys/dev/fdt/simplefb.c	6 Oct 2019 21:05:10 -0000
@@ -72,6 +72,9 @@ struct simplefb_softc {
 	psize_t			sc_psize;
 };
 
+int			(*simplefb_read_cmd)(u_long, caddr_t);
+int			(*simplefb_write_cmd)(u_long, caddr_t);
+
 struct rasops_info simplefb_ri;
 struct wsscreen_descr simplefb_wsd = { "std" };
 struct wsdisplay_charcell simplefb_bs[SIMPLEFB_WIDTH * SIMPLEFB_HEIGHT];
@@ -230,9 +233,32 @@ int
 simplefb_wsioctl(void *v, u_long cmd, caddr_t data, int flag, struct proc *p)
 {
 	struct rasops_info *ri = v;
+	struct wsdisplay_param *dp = (struct wsdisplay_param *)data;
 	struct wsdisplay_fbinfo	*wdf;
 
 	switch (cmd) {
+	case WSDISPLAYIO_GETPARAM:
+		switch (dp->param) {
+		case WSDISPLAYIO_PARAM_BRIGHTNESS:
+			if (simplefb_read_cmd) {
+				return simplefb_read_cmd(cmd, data);
+			}
+			break;
+		default:
+			return -1;
+		}
+		break;
+	case WSDISPLAYIO_SETPARAM:
+		switch (dp->param) {
+		case WSDISPLAYIO_PARAM_BRIGHTNESS:
+			if (simplefb_write_cmd) {
+				return simplefb_write_cmd(cmd, data);
+			}
+			break;
+		default:
+			return -1;
+		}
+		break;
 	case WSDISPLAYIO_GTYPE:
 		*(int *)data = WSDISPLAY_TYPE_EFIFB;
 		return 0;
