--- ./sys/dev/fdt/sxiccmu_clocks.h.orig	Fri Apr 12 23:23:30 2019
+++ ./sys/dev/fdt/sxiccmu_clocks.h	Fri Apr 12 23:24:57 2019
@@ -140,9 +140,12 @@ struct sxiccmu_ccu_bit sun8i_a23_gates[] = {
 
 /* A64 */
 
+#define A64_CLK_PLL_CPUX	1
+
 #define A64_CLK_PLL_PERIPH0	11
 #define A64_CLK_PLL_PERIPH0_2X	12
 
+#define A64_CLK_CPUX		21
 #define A64_CLK_AXI		22
 #define A64_CLK_APB		23
 #define A64_CLK_AHB1		24
