Index: sys/dev/microcode/atmel/Makefile
===================================================================
RCS file: /cvs/src/sys/dev/microcode/atmel/Makefile,v
retrieving revision 1.26
diff -u -p -r1.26 Makefile
--- sys/dev/microcode/atmel/Makefile	3 Oct 2021 00:48:45 -0000	1.26
+++ sys/dev/microcode/atmel/Makefile	2 Feb 2024 22:34:32 -0000
@@ -3,6 +3,7 @@
 NOPROG=
 NOMAN=
 CC=${HOSTCC}
+CFLAGS:=
 
 FIRM= 	atu-intersil-int atu-intersil-ext \
 	atu-rfmd-int atu-rfmd-ext \
