Index: sys/dev/microcode/yds/Makefile
===================================================================
RCS file: /cvs/src/sys/dev/microcode/yds/Makefile,v
retrieving revision 1.7
diff -u -p -r1.7 Makefile
--- sys/dev/microcode/yds/Makefile	3 Oct 2021 00:48:46 -0000	1.7
+++ sys/dev/microcode/yds/Makefile	2 Feb 2024 22:34:57 -0000
@@ -3,6 +3,7 @@
 NOPROG=
 NOMAN=
 CC=${HOSTCC}
+CFLAGS:=
 
 FIRM= 	yds
 
