Index: sys/arch/arm/arm/exception.S
===================================================================
RCS file: /cvs/src/sys/arch/arm/arm/exception.S,v
retrieving revision 1.11
diff -u -p -r1.11 exception.S
--- sys/arch/arm/arm/exception.S	8 Dec 2022 01:25:44 -0000	1.11
+++ sys/arch/arm/arm/exception.S	2 Feb 2024 22:34:09 -0000
@@ -93,8 +93,8 @@ ASENTRY_NP(swi_entry)
 	DO_AST
 	PULLFRAME
 	movs	pc, lr			/* Exit */
-	dsb	nsh
-	isb
+//	dsb	nsh
+//	isb
 
 /*
  * prefetch_abort_entry:
