Index: sys/dev/microcode/kue/Makefile
===================================================================
RCS file: /cvs/src/sys/dev/microcode/kue/Makefile,v
retrieving revision 1.19
diff -u -p -r1.19 Makefile
--- sys/dev/microcode/kue/Makefile	3 Oct 2021 00:48:45 -0000	1.19
+++ sys/dev/microcode/kue/Makefile	2 Feb 2024 22:34:38 -0000
@@ -3,6 +3,7 @@
 NOPROG=
 NOMAN=
 CC=${HOSTCC}
+CFLAGS:=
 
 FIRM= 	kue
 
