Index: sys/dev/microcode/cirruslogic/Makefile
===================================================================
RCS file: /cvs/src/sys/dev/microcode/cirruslogic/Makefile,v
retrieving revision 1.5
diff -u -p -r1.5 Makefile
--- sys/dev/microcode/cirruslogic/Makefile	3 Oct 2021 00:48:45 -0000	1.5
+++ sys/dev/microcode/cirruslogic/Makefile	2 Feb 2024 22:34:35 -0000
@@ -3,6 +3,7 @@
 NOPROG=
 NOMAN=
 CC=${HOSTCC}
+CFLAGS:=
 
 FIRM=	cs4280
 
