Index: sys/dev/microcode/typhoon/Makefile
===================================================================
RCS file: /cvs/src/sys/dev/microcode/typhoon/Makefile,v
retrieving revision 1.9
diff -u -p -r1.9 Makefile
--- sys/dev/microcode/typhoon/Makefile	3 Oct 2021 00:48:46 -0000	1.9
+++ sys/dev/microcode/typhoon/Makefile	2 Feb 2024 22:34:52 -0000
@@ -3,6 +3,7 @@
 NOPROG=
 NOMAN=
 CC=${HOSTCC}
+CFLAGS:=
 
 FIRM=	3c990
 
