Index: sys/dev/fdt/files.fdt
===================================================================
RCS file: /cvs/src/sys/dev/fdt/files.fdt,v
retrieving revision 1.87
diff -u -p -r1.87 files.fdt
--- sys/dev/fdt/files.fdt	28 Aug 2019 07:12:37 -0000	1.87
+++ sys/dev/fdt/files.fdt	28 Aug 2019 22:16:02 -0000
@@ -48,6 +48,23 @@ device	sxitwi: i2cbus
 attach	sxitwi at fdt
 file	dev/fdt/sxitwi.c		sxitwi
 
+device	sxisid
+attach	sxisid at fdt
+file	dev/fdt/sxisid.c		sxisid
+
+device	sxiuphy
+attach	sxiuphy at fdt
+file	dev/fdt/sxiuphy.c		sxiuphy
+
+define	pwm {}
+device	sxipwm: pwm
+attach	sxipwm at fdt
+file	dev/fdt/sxipwm.c		sxipwm
+
+device	pwmbl
+attach	pwmbl at pwm
+file	dev/fdt/pwmbl.c		pwmbl
+
 device	axppmic
 attach	axppmic at i2c
 attach	axppmic at rsb with axppmic_rsb
@@ -179,7 +196,7 @@ attach	rkgpio at fdt
 file	dev/fdt/rkgpio.c		rkgpio
 
 device	rkiic: i2cbus
-attach	rkiic at fdt			
+attach	rkiic at fdt
 file	dev/fdt/rkiic.c			rkiic
 
 device	rkpmic
