Index: sys/arch/armv7/armv7/armv7_start.S
===================================================================
RCS file: /cvs/src/sys/arch/armv7/armv7/armv7_start.S,v
retrieving revision 1.17
diff -u -p -r1.17 armv7_start.S
--- sys/arch/armv7/armv7/armv7_start.S	6 May 2019 03:32:48 -0000	1.17
+++ sys/arch/armv7/armv7/armv7_start.S	2 Feb 2024 22:34:26 -0000
@@ -39,8 +39,9 @@
 
 #ifdef __clang__
 .arch_extension sec
-.arch_extension virt
-#define HVC	hvc	#0
+//.arch_extension virt
+//#define HVC	hvc	#0
+#define HVC	.word	0xe1400070		@ hvc
 #else
 #define HVC	.word	0xe1400070		@ hvc
 #endif
@@ -107,6 +108,6 @@ ENTRY(omap4_smc_call)
 	stmfd   sp!, {r2-r12, lr}
 	mov     r12, r0
 	mov     r0, r1
-	dsb	sy
+//	dsb	sy
 	smc     #0
 	ldmfd   sp!, {r2-r12, pc}
