Index: sys/arch/arm64/conf/RAMDISK
===================================================================
RCS file: /cvs/src/sys/arch/arm64/conf/RAMDISK,v
retrieving revision 1.103
diff -u -p -r1.103 RAMDISK
--- sys/arch/arm64/conf/RAMDISK	9 Sep 2019 20:00:27 -0000	1.103
+++ sys/arch/arm64/conf/RAMDISK	6 Oct 2019 09:51:24 -0000
@@ -107,6 +107,7 @@ vioscsi*	at virtio?
 
 simplefb*	at fdt?
 wsdisplay*	at simplefb?
+pwmbl*		at pwm?	# PWM backlight
 
 # iMX
 imxccm*		at fdt? early 1
@@ -186,6 +187,8 @@ sximmc*		at fdt?		# SD/MMC card controll
 sdmmc*		at sximmc?	# SD/MMC bus
 sxisyscon*	at fdt? early 1	# System controller
 sxitwi*		at fdt?		# I2C controller
+sxipwm*		at fdt?		# PWM controller
+sxiuphy*	at fdt?		# USB Phy
 iic*		at sxitwi?	# I2C bus
 dwxe*		at fdt?
 
