Index: sys/arch/arm64/conf/GENERIC
===================================================================
RCS file: /cvs/src/sys/arch/arm64/conf/GENERIC,v
retrieving revision 1.126
diff -u -p -r1.126 GENERIC
--- sys/arch/arm64/conf/GENERIC	30 Sep 2019 20:48:15 -0000	1.126
+++ sys/arch/arm64/conf/GENERIC	6 Oct 2019 21:04:59 -0000
@@ -105,6 +105,7 @@ option	WSDISPLAY_DEFAULTSCREENS=6	# init
 
 simplefb*	at fdt?
 wsdisplay*	at simplefb?
+pwmbl*			at fdt?		# PWM backlight
 
 radeondrm*	at pci?
 drm*		at radeondrm?
@@ -203,6 +204,7 @@ sdmmc*		at sximmc?	# SD/MMC bus
 sxisyscon*	at fdt? early 1	# System controller
 sxitemp*	at fdt?		# Temperature sensor
 sxitwi*		at fdt?		# I2C controller
+sxipwm*		at fdt?	# PWM controller
 iic*		at sxitwi?	# I2C bus
 dwxe*		at fdt?
 
