Index: sys/arch/arm/arm/locore.S
===================================================================
RCS file: /cvs/src/sys/arch/arm/arm/locore.S,v
retrieving revision 1.23
diff -u -p -r1.23 locore.S
--- sys/arch/arm/arm/locore.S	8 Dec 2022 01:25:44 -0000	1.23
+++ sys/arch/arm/arm/locore.S	2 Feb 2024 22:34:12 -0000
@@ -46,7 +46,7 @@
 /* What size should this really be ? It is only used by init_arm() */
 #define INIT_ARM_STACK_SIZE	2048
 
-.arch_extension virt
+//.arch_extension virt
 
 /*
  * This is for kvm_mkdb, and should be the address of the beginning
@@ -119,17 +119,17 @@ drop_to_pl1:
 	orr	r0, r0, #(PSR_SVC32_MODE)
 	orr	r0, r0, #(PSR_I | PSR_F | PSR_A)
 	msr	spsr_fsxc, r0
-	msr	ELR_hyp, lr
-	eret				/* Use the HYP LR to get into SVC */
-	dsb nsh
-	isb
+//	msr	ELR_hyp, lr
+//	eret				/* Use the HYP LR to get into SVC */
+//	dsb nsh
+//	isb
 
 	.globl start_mmu
 start_mmu:
 	/* Set ASID to zero */
 	mov	r1, #0
 	mcr	CP15_CONTEXTIDR(r1)
-	isb
+//	isb
 
 	mcr	CP15_TTBR0(r0)		/* Set TTB */
 	mcr	CP15_TLBIALL(r0)	/* Flush TLB */
@@ -137,13 +137,13 @@ start_mmu:
 	/* Set the Domain Access register.  Very important! */
 	mov     r0, #DOMAIN_CLIENT	/* We only use domain 0 */
 	mcr	CP15_DACR(r0)
-	isb
+//	isb
 
 	/* Enable MMU */
 	mrc	CP15_SCTLR(r0)
 	orr	r0, r0, #CPU_CONTROL_MMU_ENABLE
 	mcr	CP15_SCTLR(r0)
-	isb
+//	isb
 
 	mov	pc, lr
 
