Index: sys/arch/armv7/armv7/platform.c
===================================================================
RCS file: /cvs/src/sys/arch/armv7/armv7/platform.c,v
retrieving revision 1.27
diff -u -p -r1.27 platform.c
--- sys/arch/armv7/armv7/platform.c	16 May 2021 03:39:28 -0000	1.27
+++ sys/arch/armv7/armv7/platform.c	14 Feb 2024 11:29:59 -0000
@@ -33,6 +33,7 @@ void	agtimer_init(void);
 extern void	cduart_init_cons(void);
 extern void	exuart_init_cons(void);
 extern void	imxuart_init_cons(void);
+extern void     stmuart_init_cons(void);
 extern void	com_fdt_init_cons(void);
 extern void	pluart_init_cons(void);
 extern void	simplefb_init_cons(bus_space_tag_t);
@@ -91,6 +92,7 @@ platform_init_cons(void)
 	cduart_init_cons();
 	exuart_init_cons();
 	imxuart_init_cons();
+	stmuart_init_cons();
 	com_fdt_init_cons();
 	pluart_init_cons();
 	simplefb_init_cons(&armv7_bs_tag);
