Index: sys/arch/arm/include/armreg.h
===================================================================
RCS file: /cvs/src/sys/arch/arm/include/armreg.h,v
retrieving revision 1.43
diff -u -p -r1.43 armreg.h
--- sys/arch/arm/include/armreg.h	30 Sep 2019 21:48:32 -0000	1.43
+++ sys/arch/arm/include/armreg.h	2 Feb 2024 22:34:21 -0000
@@ -161,6 +161,8 @@
 #define CPU_ID_CORTEX_MASK	0xff0ffff0
 #define CPU_ID_CORTEX_A5	0x410fc050
 #define CPU_ID_CORTEX_A5_MASK	0xff0ffff0
+#define CPU_ID_CORTEX_A6        0x410fc060
+#define CPU_ID_CORTEX_A6_MASK   0xff0ffff0
 #define CPU_ID_CORTEX_A7	0x410fc070
 #define CPU_ID_CORTEX_A7_MASK	0xff0ffff0
 #define CPU_ID_CORTEX_A8_R1	0x411fc080
