Index: sys/dev/microcode/tusb3410/Makefile
===================================================================
RCS file: /cvs/src/sys/dev/microcode/tusb3410/Makefile,v
retrieving revision 1.13
diff -u -p -r1.13 Makefile
--- sys/dev/microcode/tusb3410/Makefile	3 Oct 2021 00:48:45 -0000	1.13
+++ sys/dev/microcode/tusb3410/Makefile	2 Feb 2024 22:34:51 -0000
@@ -3,6 +3,7 @@
 NOPROG=
 NOMAN=
 CC=${HOSTCC}
+CFLAGS:=
 
 FIRM= 	tusb3410
 
