Index: sys/arch/arm64/conf/GENERIC
===================================================================
RCS file: /cvs/src/sys/arch/arm64/conf/GENERIC,v
retrieving revision 1.131
diff -u -p -r1.131 GENERIC
--- sys/arch/arm64/conf/GENERIC	21 Oct 2019 20:52:32 -0000	1.131
+++ sys/arch/arm64/conf/GENERIC	23 Oct 2019 01:44:24 -0000
@@ -208,8 +208,10 @@ sxirtc*		at fdt? early 1	# Real Time Clo
 sximmc*		at fdt?		# SD/MMC card controller
 sdmmc*		at sximmc?	# SD/MMC bus
 sxisyscon*	at fdt? early 1	# System controller
+sxisid*		at fdt? early 1	# SID ROM
 sxitemp*	at fdt?		# Temperature sensor
 sxitwi*		at fdt?		# I2C controller
+sxiuphy*	at fdt?		# USB Phy
 iic*		at sxitwi?	# I2C bus
 dwxe*		at fdt?
 
