Index: sys/dev/fdt/files.fdt
===================================================================
RCS file: /cvs/src/sys/dev/fdt/files.fdt,v
retrieving revision 1.95
diff -u -p -r1.95 files.fdt
--- sys/dev/fdt/files.fdt	30 Sep 2019 20:44:13 -0000	1.95
+++ sys/dev/fdt/files.fdt	1 Oct 2019 18:58:51 -0000
@@ -48,6 +48,10 @@ device	sxitwi: i2cbus
 attach	sxitwi at fdt
 file	dev/fdt/sxitwi.c		sxitwi
 
+device	sxiuphy
+attach	sxiuphy at fdt
+file	dev/fdt/sxiuphy.c		sxiuphy
+
 device	axppmic
 attach	axppmic at i2c
 attach	axppmic at rsb with axppmic_rsb
