Index: sys/arch/armv7/conf/Makefile.armv7
===================================================================
RCS file: /cvs/src/sys/arch/armv7/conf/Makefile.armv7,v
retrieving revision 1.55
diff -u -p -r1.55 Makefile.armv7
--- sys/arch/armv7/conf/Makefile.armv7	28 Jan 2024 00:40:22 -0000	1.55
+++ sys/arch/armv7/conf/Makefile.armv7	2 Feb 2024 22:34:29 -0000
@@ -29,7 +29,7 @@ CWARNFLAGS=	-Werror -Wall -Wimplicit-fun
 		-Wno-unused-but-set-variable -Wno-gnu-folding-constant \
 		-Wframe-larger-than=2047
 
-CMACHFLAGS=	-msoft-float -march=armv7a
+CMACHFLAGS=	-msoft-float -march=armv6
 CMACHFLAGS+=	-ffreestanding ${NOPIE_FLAGS}
 SORTR=		sort -R
 .if ${IDENT:M-DNO_PROPOLICE}
@@ -40,10 +40,12 @@ SORTR=		cat
 COPTIMIZE?=	-Oz
 .endif
 
-DEBUG?=		-g
+DEBUG?=		-g -O0
 COPTIMIZE?=	-O2
-CFLAGS=		${DEBUG} ${CWARNFLAGS} ${CMACHFLAGS} ${COPTIMIZE} ${COPTS} ${PIPE}
-AFLAGS=		-D_LOCORE -x assembler-with-cpp ${CWARNFLAGS} ${CMACHFLAGS}
+CFLAGS=		${DEBUG} ${CWARNFLAGS} ${CMACHFLAGS} ${COPTIMIZE} ${COPTS} ${PIPE} \
+    -march=armv6
+    #-marm -mfpu=vfp -mcpu=arm1176jzf-s -mtune=arm1176jzf-s -mfloat-abi=softfp
+AFLAGS=		-D_LOCORE -x assembler-with-cpp ${CWARNFLAGS} ${CMACHFLAGS} -march=armv6 -DCPU_ARMv6
 LINKFLAGS=	-T ld.script --warn-common -nopie
 
 HOSTCC?=	${CC}
