Index: usr.bin/dig/Makefile
===================================================================
RCS file: /cvs/src/usr.bin/dig/Makefile,v
retrieving revision 1.4
diff -u -p -r1.4 Makefile
--- usr.bin/dig/Makefile	14 Sep 2020 08:39:12 -0000	1.4
+++ usr.bin/dig/Makefile	2 Feb 2024 22:35:00 -0000
@@ -13,7 +13,7 @@ CLEANFILES+=	code.h gen gen.d
 .include "${.CURDIR}/lib/isccfg/Makefile.inc"
 .include "${.CURDIR}/lib/lwres/Makefile.inc"
 
-CFLAGS+= -Wall -I${.CURDIR} -I${.OBJDIR}
+CFLAGS:= -Wall -I${.CURDIR} -I${.OBJDIR}
 CFLAGS+= -I ${.CURDIR}/bin/dig/include
 CFLAGS+= -I ${.CURDIR}/lib/dns/include
 CFLAGS+= -I ${.CURDIR}/lib/isc/include
@@ -31,7 +31,7 @@ DPADD+= ${LIBCRYPTO}
 BUILDFIRST = code.h
 
 gen: lib/dns/gen.c
-	${HOSTCC} ${CFLAGS} ${.CURDIR}/lib/dns/gen.c -o $@
+	${HOSTCC} ${.CURDIR}/lib/dns/gen.c -o $@
 
 code.h:	gen
 	./gen -s ${.CURDIR}/lib/dns > code.h
